Data/Primitive/ByteArray.hs

```diff
@@ -98,6 +100,9 @@ data ByteArray = ByteArray ByteArray# deriving ( Typeable )
 data MutableByteArray s = MutableByteArray (MutableByteArray# s)
                                         deriving( Typeable )
 
+instance NFData ByteArray where
+  rnf = rwhnf
```

Comment:
`rwnf` is only available since `deepseq-1.4.3.0`, which started to be shipped with gch-8.2.

I suggest a simple pattern match:
```haskell
instance NFData ByteArray where
  rnf (ByteArray _) = ()
```

I could also recommend adding one for `MutableByteArray`, which will work as intended since all elements must always be in normal form:

```haskell
instance NFData (MutableByteArray s) where
  rnf (MutableByteArray _) = ()
```

Same thing applies to `PrimArray` and `MutablePrimArray`

Reply:
I agree. This also makes it much more clear that it is correct.